File: Ambulance.BLL/Models/CallAnalyticsModel.cs
namespace Ambulance.BLL.Models;

public class CallAnalyticsModel
{
    public int TotalCalls { get; set; }
    public int CompletedCalls { get; set; }
    public double AverageResponseMinutes { get; set; }
    public Dictionary<string, int> CallsByUrgency { get; set; } = new(); // тип терміновості та кількість викликів
}

File: ambulance-site/src/api/models/AnaliticsModels/CallAnalyticsModel.js
// models/callAnalyticsModel.js
export class CallAnalyticsModel {
    constructor({ totalCalls = 0, completedCalls = 0, averageResponseMinutes = 0.0, callsByUrgency = {} } = {}) {
        this.totalCalls = totalCalls;
        this.completedCalls = completedCalls;
        this.averageResponseMinutes = averageResponseMinutes;
        this.callsByUrgency = callsByUrgency; // об'єкт у форматі {urgencyType: count}
    }
}
